Content.Server/Supermatter/Components/SupermatterComponent.cs

```diff
@@ -0,0 +1,167 @@
+        //Higher == More moles of the gas are needed before the charge inertia chain reaction effect starts.
+        //Scales powerloss inhibition down until this amount of moles is reached
+        public const float  PowerlossInhibitionMoleThreshold = 20f;
+        //bonus powerloss inhibition boost if this amount of moles is reached
+        public const float  PowerlossInhibitionMoleBoostThreshold = 500f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //Above this value we can get lord singulo and independent mol damage, below it we can heal damage
+        public const float  MolePenaltyThreshold = 1800f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //Along with damagepenaltypoint, makes flux anomalies.
+        /// The cutoff for the minimum amount of power required to trigger the crystal invasion delamination event.
+        public const float  EventPowerPenaltyThreshold = 4500f;
+        //The cutoff on power properly doing damage, pulling shit around, and delamming into a tesla. Low chance of pyro anomalies, +2 bolts of electricity
+        public const float  PowerPenaltyThreshold = 5000f;
+        //+1 bolt of electricity, allows for gravitational anomalies, and higher chances of pyro anomalies
+        public const float  SeverePowerPenaltyThreshold = 7000f;
+        //+1 bolt of electricity.
+        public const float  CriticalPowerPenaltyThreshold = 9000f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //Higher == Crystal safe operational temperature is higher.
+        public const float  HeatPenaltyThreshold = 40f;
+        //is multiplied by ExplosionPoint to cap evironmental damage per cycle
+        public const float  DamageHardcap = 0.002f;
+        //environmental damage is scaled by this
+        public const float  DamageIncreaseMultiplier = 0.25f;
+        //if spaced sm wont take more than 2 damage per cycle
+        public const float MaxSpaceExposureDamage = 2;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //Higher == more overall power
+        public const float  ReactionPowerModefier = 0.55f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //These would be what you would get at point blank, decreases with distance
+        public const float  DetonationRads = 200f;
+        public const float  DetonationHallucination = 600f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        [ViewVariables(VVAccess.ReadOnly)]
+        public float[] GasComp =
+        {
+            0, //oxy
+            0, //Nit
+            0, //Co2
+            0, //pla
+            0, //tri
+            0, //h2o
+        };
```

Comment:
This shouldn't be saved like that. Use the `GetMoles(Gas gas)` method in the `SupermatterSystem`.
*See my other comment*